File: sources/theta-app/src/app/models/itinerary.ts
import { Pace } from '@theta/enums/pace';
import { Activity } from './activity';
import { BaseModel } from './base-model';
import { City } from './city';
import { Price } from './price';

export interface Itinerary extends BaseModel {
    name: string;
    score: string;
    activities: Activity[];
    city: City;
    estimatedBudget: Price;
    description: string;
    tags: string[];
    pace: Pace;
    userId: string;
    imageUrls: string[];
}

File: api/Theta.Core/Entities/ItineraryAggregate/Itinerary.cs
public class Itinerary : BaseEntity, IAggregatedRoot
{
    public string Name { get; set; } = null!;
    public float? Score { get; set; }
    public decimal? EstimatedBudget { get; set; }
    public string? Description { get; set; }
    public IEnumerable<Activity> Activities { get; set; } = Enumerable.Empty<Activity>();
}
